src/main/java/com/puppycrawl/tools/checkstyle/checks/header/AbstractHeaderCheck.java

```diff
@@ -46,7 +47,7 @@
  * @author o_sukhosolsky
  */
 public abstract class AbstractHeaderCheck extends AbstractFileSetCheck
-    implements ExternalResourceHolder {
+    implements ExternalResourceHolder, OneModuleInstancePerApplication {
```

Comment:
Marking abstract classes is a risky business.
`OneModuleInstancePerApplication` is the weakest type to apply to a module. What if a module that implements this abstract class requires the strongest type, like `OneModuleInstancePerThread`.
You will now have a conflict and the module will implement both interfaces.

IMO, we either shouldn't mark abstract classes or if the define both, `OneModuleInstancePerThread` takes precedence.